skorch/tests/test_net.py

```diff
@@ -2540,6 +2540,92 @@ def forward(self, y_pred, _):
         with raises:
             net.fit(X, y)
 
+    def test_predict_nonlinearity_called_with_predict(
+            self, net_cls, module_cls, data):
+        side_effect = []
+        def nonlin(X):
+            side_effect.append(X)
+            return np.zeros_like(X)
+
+        X, y = data[0][:200], data[1][:200]
+        net = net_cls(
+            module_cls, max_epochs=1, predict_nonlinearity=nonlin).initialize()
+
+        # don't want callbacks to trigger side effects
+        net.callbacks_ = []
+        net.partial_fit(X, y)
+        assert not side_effect
```

Comment:
But wouldn't we expect callbacks such as accuracy scoring to call `predict`?

Reply:
That's why I removed all callbacks two lines earlier, otherwise the test becomes very messy.